### PR TITLE
Structured Logging migration: modify logs for kinds of kuberentes volumes

### PR DESCRIPTION
**Pull request by wangyx1992**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Structured Logging migration: modify logs for kinds of kuberentes volumes.
xref:
https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md

#### Which issue(s) this PR fixes:
Fixes #

#### Special notes for your reviewer:
part of kubernetes/enhancements#1602
As it is related to log, change the log here to structured log.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: pkg/volume/awsebs/aws_util.go [hunk 1]
```diff
@@ -232,14 +232,14 @@ func getDiskByIDPaths(volumeID aws.KubernetesVolumeID, partition string, deviceP
...
-		klog.Warningf("error mapping volume %q to AWS volume: %v", volumeID, err)
+		klog.InfoS("Error mapping volume to AWS volume", "volumeID", volumeID, "err", err)
 	} else {
 		// This is the magic name on which AWS presents NVME devices under /dev/disk/by-id/
 		// For example, vol-0fab1d5e3f72a5e23 creates a symlink at /dev/disk/by-id/nvme-Amazon_Elastic_Block_Store_vol0fab1d5e3f72a5e23
 		nvmeName := "nvme-Amazon_Elastic_Block_Store_" + strings.Replace(string(awsVolumeID), "-", "", -1)
 		nvmePath, err := findNvmeVolume(nvmeName)
 		if err != nil {
-			klog.Warningf("error looking for nvme volume %q: %v", volumeID, err)
+			klog.InfoS("Error looking for nvme volume ", "volumeID", volumeID, "volumeID", err)
```
Comment (yangjunmyfm192085) [inline pkg/volume/awsebs/aws_util.go:242]:
```suggestion
			klog.InfoS("Error looking for nvme volume ", "volumeID", volumeID, "err", err)
```

File: pkg/volume/vsphere_volume/attacher.go [hunk 2]
```diff
@@ -80,7 +80,7 @@ func (attacher *vsphereVMDKAttacher) Attach(spec *volume.Spec, nodeName types.No
 		return "", err
 	}
 
-	klog.V(4).Infof("vSphere: Attach disk called for node %s", nodeName)
+	klog.V(4).InfoS("vSphere: Attach disk called for node", "nodeName", nodeName)
```
Comment (yangjunmyfm192085) [inline pkg/volume/vsphere_volume/attacher.go:83]:
```suggestion
	klog.V(4).InfoS("VSphere: Attach disk called for node", "nodeName", nodeName)
```

File: pkg/volume/vsphere_volume/attacher.go [hunk 3]
```diff
@@ -209,7 +209,7 @@ func (plugin *vsphereVolumePlugin) GetDeviceMountRefs(deviceMountPath string) ([
 
 // MountDevice mounts device to global mount point.
 func (attacher *vsphereVMDKAttacher) MountDevice(spec *volume.Spec, devicePath string, deviceMountPath string) error {
-	klog.Infof("vsphere MountDevice mount %s to %s", devicePath, deviceMountPath)
+	klog.InfoS("vSphere MountDevice do mount", "fromPath", devicePath, "mountPath", deviceMountPath)
```
Comment (yangjunmyfm192085) [inline pkg/volume/vsphere_volume/attacher.go:212]:
```suggestion
	klog.InfoS("VSphere MountDevice do mount", "fromPath", devicePath, "mountPath", deviceMountPath)
```

File: pkg/volume/awsebs/aws_util.go (hunk 1 above)
Comment (wangyx1992) [reply pkg/volume/awsebs/aws_util.go]: Thanks for your review, the clerical error has been fixed. @yangjunmyfm192085

File: pkg/volume/vsphere_volume/attacher.go (hunk 2 above)
Comment (wangyx1992) [reply pkg/volume/vsphere_volume/attacher.go:83]: Thanks for your suggestion, but vSphere probably is the custom abbreviation.

**Comment by wangyx1992** [comment]
/assign @serathius 
According to [https://github.com/kubernetes/enhancements/issues/1602](url), this pr modifies some logs in volume.
#101088 #101095 #100672

File: pkg/volume/vsphere_volume/attacher.go [hunk 4]
```diff
@@ -279,21 +279,21 @@ func (detacher *vsphereVMDKDetacher) Detach(volumeName string, nodeName types.No
...
-		klog.Infof("detach operation was successful. volume %q is already detached from node %q.", volPath, nodeName)
+		klog.InfoS("Detach operation was successful. volume is already detached from node.", "volumePath", volPath, "nodeName", nodeName)
 		return nil
 	}
 
 	attachdetachMutex.LockKey(string(nodeName))
 	defer attachdetachMutex.UnlockKey(string(nodeName))
 	if err := detacher.vsphereVolumes.DetachDisk(newVolumePath, nodeName); err != nil {
-		klog.Errorf("Error detaching volume %q: %v", volPath, err)
+		klog.ErrorS(err, "Error detaching volume", "volumePath", volPath)
```
Comment (serathius) [inline pkg/volume/vsphere_volume/attacher.go:296]:
```suggestion
		klog.ErrorS(err, "Error detaching volume", "path", volPath)
```

File: pkg/volume/vsphere_volume/attacher.go [hunk 5]
```diff
@@ -279,21 +279,21 @@ func (detacher *vsphereVMDKDetacher) Detach(volumeName string, nodeName types.No
...
-			volPath, nodeName, err)
+		klog.ErrorS(err,
+			"Error checking if volume is already attached to current node. Will continue and try detach anyway.",
+			"volumePath", volPath, "nodeName", nodeName)
 	}
 
 	if err == nil && !attached {
 		// Volume is already detached from node.
-		klog.Infof("detach operation was successful. volume %q is already detached from node %q.", volPath, nodeName)
+		klog.InfoS("Detach operation was successful. volume is already detached from node.", "volumePath", volPath, "nodeName", nodeName)
```
Comment (serathius) [inline pkg/volume/vsphere_volume/attacher.go:289]:
```suggestion
		klog.InfoS("Detach operation was successful. volume is already detached from node.", "path", volPath, "nodeName", nodeName)
```

File: pkg/volume/vsphere_volume/attacher.go [hunk 6]
```diff
@@ -173,14 +173,14 @@ func (attacher *vsphereVMDKAttacher) WaitForAttach(spec *volume.Spec, devicePath
...
+			klog.V(5).InfoS("Checking VMDK is attached", "volumePath", volumeSource.VolumePath)
 			path, err := verifyDevicePath(devicePath)
 			if err != nil {
 				// Log error, if any, and continue checking periodically. See issue #11321
-				klog.Warningf("Error verifying VMDK (%q) is attached: %v", volumeSource.VolumePath, err)
+				klog.InfoS("Error verifying VMDK is attached", "volumePath", volumeSource.VolumePath, "err", err)
 			} else if path != "" {
 				// A device path has successfully been created for the VMDK
-				klog.Infof("Successfully found attached VMDK %q.", volumeSource.VolumePath)
+				klog.InfoS("Successfully found attached VMDK.", "volumePath", volumeSource.VolumePath)
```
Comment (serathius) [inline pkg/volume/vsphere_volume/attacher.go:183]:
```suggestion
				klog.InfoS("Successfully found attached VMDK.", "path", volumeSource.VolumePath)
```

File: pkg/volume/vsphere_volume/attacher.go [hunk 7]
```diff
@@ -173,14 +173,14 @@ func (attacher *vsphereVMDKAttacher) WaitForAttach(spec *volume.Spec, devicePath
 	for {
 		select {
 		case <-ticker.C:
-			klog.V(5).Infof("Checking VMDK %q is attached", volumeSource.VolumePath)
+			klog.V(5).InfoS("Checking VMDK is attached", "volumePath", volumeSource.VolumePath)
 			path, err := verifyDevicePath(devicePath)
 			if err != nil {
 				// Log error, if any, and continue checking periodically. See issue #11321
-				klog.Warningf("Error verifying VMDK (%q) is attached: %v", volumeSource.VolumePath, err)
+				klog.InfoS("Error verifying VMDK is attached", "volumePath", volumeSource.VolumePath, "err", err)
```
Comment (serathius) [inline pkg/volume/vsphere_volume/attacher.go:180]:
```suggestion
				klog.InfoS("Error verifying VMDK is attached", "path", volumeSource.VolumePath, "err", err)
```

File: pkg/volume/vsphere_volume/attacher.go [hunk 8]
```diff
@@ -173,14 +173,14 @@ func (attacher *vsphereVMDKAttacher) WaitForAttach(spec *volume.Spec, devicePath
 	for {
 		select {
 		case <-ticker.C:
-			klog.V(5).Infof("Checking VMDK %q is attached", volumeSource.VolumePath)
+			klog.V(5).InfoS("Checking VMDK is attached", "volumePath", volumeSource.VolumePath)
```
Comment (serathius) [inline pkg/volume/vsphere_volume/attacher.go:176]:
```suggestion
			klog.V(5).InfoS("Checking VMDK is attached", "path", volumeSource.VolumePath)
```

File: pkg/volume/vsphere_volume/attacher.go [hunk 9]
```diff
@@ -90,22 +90,22 @@ func (attacher *vsphereVMDKAttacher) Attach(spec *volume.Spec, nodeName types.No
...
-	klog.Warningf("Attacher.VolumesAreAttached called for node %q - Please use BulkVerifyVolumes for vSphere", nodeName)
+	klog.InfoS("Attacher.VolumesAreAttached called for node - Please use BulkVerifyVolumes for vSphere", "nodeName", nodeName)
 	volumeNodeMap := map[types.NodeName][]*volume.Spec{
 		nodeName: specs,
 	}
 	nodeVolumesResult := make(map[*volume.Spec]bool)
 	nodesVerificationMap, err := attacher.BulkVerifyVolumes(volumeNodeMap)
 	if err != nil {
-		klog.Errorf("Attacher.VolumesAreAttached - error checking volumes for node %q with %v", nodeName, err)
+		klog.ErrorS(err, "Attacher.VolumesAreAttached - error checking volumes for node", "nodeName", nodeName)
```
Comment (serathius) [inline pkg/volume/vsphere_volume/attacher.go:108]:
```suggestion
		klog.ErrorS(err, "Attacher.VolumesAreAttached - error checking volumes for node", "node", klog.KRef("", nodeName))
```

File: pkg/volume/vsphere_volume/attacher.go [hunk 10]
```diff
@@ -90,22 +90,22 @@ func (attacher *vsphereVMDKAttacher) Attach(spec *volume.Spec, nodeName types.No
...
+		klog.ErrorS(err, "Error attaching volume to node", "volumePath", volumeSource.VolumePath, "nodeName", nodeName)
 		return "", err
 	}
 
 	return filepath.Join(diskByIDPath, diskSCSIPrefix+diskUUID), nil
 }
 
 func (attacher *vsphereVMDKAttacher) VolumesAreAttached(specs []*volume.Spec, nodeName types.NodeName) (map[*volume.Spec]bool, error) {
-	klog.Warningf("Attacher.VolumesAreAttached called for node %q - Please use BulkVerifyVolumes for vSphere", nodeName)
+	klog.InfoS("Attacher.VolumesAreAttached called for node - Please use BulkVerifyVolumes for vSphere", "nodeName", nodeName)
```
Comment (serathius) [inline pkg/volume/vsphere_volume/attacher.go:101]:
```suggestion
	klog.InfoS("Attacher.VolumesAreAttached called for node - Please use BulkVerifyVolumes for vSphere", "node", klog.KRef("", nodeName))
```

File: pkg/volume/vsphere_volume/attacher.go [hunk 11]
```diff
@@ -90,22 +90,22 @@ func (attacher *vsphereVMDKAttacher) Attach(spec *volume.Spec, nodeName types.No
 	// succeeds in that case, so no need to do that separately.
 	diskUUID, err := attacher.vsphereVolumes.AttachDisk(volumeSource.VolumePath, volumeSource.StoragePolicyName, nodeName)
 	if err != nil {
-		klog.Errorf("Error attaching volume %q to node %q: %+v", volumeSource.VolumePath, nodeName, err)
+		klog.ErrorS(err, "Error attaching volume to node", "volumePath", volumeSource.VolumePath, "nodeName", nodeName)
```
Comment (serathius) [inline pkg/volume/vsphere_volume/attacher.go:93]:
```suggestion
		klog.ErrorS(err, "Error attaching volume to node", "volumePath", volumeSource.VolumePath, "node", klog.KRef("", nodeName))
```

File: pkg/volume/vsphere_volume/attacher.go (hunk 2 above)
Comment (serathius) [inline pkg/volume/vsphere_volume/attacher.go:83]:
```suggestion
	klog.V(4).InfoS("vSphere: Attach disk called for node", "node", klog.KRef("", nodeName))
```

File: pkg/volume/nfs/nfs.go [hunk 12]
```diff
@@ -245,7 +245,7 @@ func (nfsMounter *nfsMounter) SetUp(mounterArgs volume.MounterArgs) error {
 
 func (nfsMounter *nfsMounter) SetUpAt(dir string, mounterArgs volume.MounterArgs) error {
 	notMnt, err := mount.IsNotMountPoint(nfsMounter.mounter, dir)
-	klog.V(4).Infof("NFS mount set up: %s %v %v", dir, !notMnt, err)
+	klog.V(4).InfoS("NFS mount set up", "path", dir, "!notMnt", !notMnt, "err", err)
```
Comment (serathius) [inline pkg/volume/nfs/nfs.go:248]: We cannot just dump variable names as we cannot assume that people that read logs need to know code.
```suggestion
	klog.V(4).InfoS("NFS mount set up", "path", dir, "isMountPoint", !notMnt, "err", err)
```

File: pkg/volume/cinder/attacher_test.go [hunk 13]
```diff
@@ -700,7 +700,7 @@ func (testcase *testcase) DisksAreAttachedByName(nodeName types.NodeName, volume
 		return areAttached, errors.New("unexpected DisksAreAttachedByName call: wrong instanceID")
 	}
 
-	klog.V(4).Infof("DisksAreAttachedByName call: %v, %s, returning %v, %v", volumeIDs, nodeName, expected.areAttached, expected.ret)
+	klog.V(4).InfoS("DisksAreAttachedByName call volumeIDs and nodeName, returns areAttached and ret", "volumeIDs", volumeIDs, "nodeName", nodeName, "areAttached", expected.areAttached, "ret", expected.ret)
```
Comment (serathius) [inline pkg/volume/cinder/attacher_test.go:703]: What is `ret` here?
```suggestion
	klog.V(4).InfoS("DisksAreAttachedByName call volumeIDs and nodeName, returns areAttached and ret", "volumeIDs", volumeIDs, "node", klog.KRef("", nodeName), "areAttached", expected.areAttached, "ret", expected.ret)
```

File: pkg/volume/azuredd/azure_common_windows.go [hunk 14]
```diff
@@ -69,27 +69,27 @@ func findDiskByLun(lun int, iohandler ioHandler, exec utilexec.Interface) (strin
...
 				l, err := strconv.Atoi(arr[arrLen-1])
 				if err != nil {
-					klog.Warningf("cannot parse element from data structure, location: %q, element: %q", location, arr[arrLen-1])
+					klog.InfoS("Cannot parse element from data structure", "location", location, "element", arr[arrLen-1])
 					continue
 				}
 
 				if l == lun {
-					klog.V(4).Infof("found a disk and lun, location: %q, lun: %d", location, lun)
+					klog.V(4).InfoS("Found a disk and lun", "location", location, "lun", lun)
```
Comment (serathius) [inline pkg/volume/azuredd/azure_common_windows.go:86]: What is `lun` here?

File: pkg/volume/azuredd/azure_common_windows.go [hunk 15]
```diff
@@ -69,27 +69,27 @@ func findDiskByLun(lun int, iohandler ioHandler, exec utilexec.Interface) (strin
...
 					if d, ok := v["number"]; ok {
 						if diskNum, ok := d.(float64); ok {
-							klog.V(2).Infof("azureDisk Mount: got disk number(%d) by LUN(%d)", int(diskNum), lun)
+							klog.V(2).InfoS("AzureDisk Mount: got disk number by LUN", "diskNum", int(diskNum), "lun", lun)
```
Comment (serathius) [inline pkg/volume/azuredd/azure_common_windows.go:89]:
```suggestion
							klog.V(2).InfoS("AzureDisk Mount: got disk number by LUN", "diskNumber", int(diskNum), "lun", lun)
```

File: pkg/volume/azuredd/azure_common_windows.go [hunk 16]
```diff
@@ -69,27 +69,27 @@ func findDiskByLun(lun int, iohandler ioHandler, exec utilexec.Interface) (strin
...
 							return fmt.Sprintf(winDiskNumFormat, int(diskNum)), nil
 						}
-						klog.Warningf("LUN(%d) found, but could not get disk number(%q), location: %q", lun, d, location)
+						klog.InfoS("LUN found, but could not get disk number in location", "lun", lun, "diskNum", d, "location", location)
```
Comment (serathius) [inline pkg/volume/azuredd/azure_common_windows.go:92]:
```suggestion
						klog.InfoS("LUN found, but could not get disk number in location", "lun", lun, "diskNumber", d, "location", location)
```

File: pkg/volume/azuredd/attacher.go [hunk 17]
```diff
@@ -276,10 +276,10 @@ func (d *azureDiskDetacher) Detach(diskURI string, nodeName types.NodeName) erro
 
 	err = diskController.DetachDisk("", diskURI, nodeName)
 	if err != nil {
-		klog.Errorf("failed to detach azure disk %q, err %v", diskURI, err)
+		klog.ErrorS(err, "Failed to detach azure disk", "diskURI", diskURI)
 	}
 
-	klog.V(2).Infof("azureDisk - disk:%s was detached from node:%v", diskURI, nodeName)
+	klog.V(2).InfoS("AzureDisk - Disk was detached from node", "diskURI", diskURI, "nodeName", nodeName)
```
Comment (serathius) [inline pkg/volume/azuredd/attacher.go:282]:
```suggestion
	klog.V(2).InfoS("AzureDisk - Disk was detached from node", "URI", diskURI, "nodeName", nodeName)
```

File: pkg/volume/azuredd/attacher.go [hunk 18]
```diff
@@ -119,17 +119,17 @@ func (a *azureDiskAttacher) VolumesAreAttached(specs []*volume.Spec, nodeName ty
 	attachedResult, err := diskController.DisksAreAttached(volumeIDList, nodeName)
 	if err != nil {
 		// Log error and continue with attach
-		klog.Errorf(
-			"azureDisk - Error checking if volumes (%v) are attached to current node (%q). err=%v",
-			volumeIDList, nodeName, err)
+		klog.ErrorS(err,
+			"AzureDisk - Error checking if volumes are attached to current node",
+			"volumeIDList", volumeIDList, "nodeName", nodeName)
```
Comment (serathius) [inline pkg/volume/azuredd/attacher.go:124]:
```suggestion
			"volumeIDs", volumeIDList, "node", klog.KRef("", nodeName))
```

File: pkg/volume/azuredd/attacher.go [hunk 19]
```diff
@@ -72,23 +72,23 @@ func (a *azureDiskAttacher) Attach(spec *volume.Spec, nodeName types.NodeName) (
...
+		klog.V(2).InfoS("Initiating attaching volume to node. GetDiskLun returned error.", "volumeDiskURI", volumeSource.DataDiskURI, "nodeName", nodeName, "err", err)
 
 		isManagedDisk := (*volumeSource.Kind == v1.AzureManagedDisk)
 		lun, err = diskController.AttachDisk(isManagedDisk, volumeSource.DiskName, volumeSource.DataDiskURI, nodeName, compute.CachingTypes(*volumeSource.CachingMode))
 		if err == nil {
-			klog.V(2).Infof("Attach operation successful: volume %q attached to node %q.", volumeSource.DataDiskURI, nodeName)
+			klog.V(2).InfoS("Attach operation successful: volume attached to node.", "volumeDiskURI", volumeSource.DataDiskURI, "nodeName", nodeName)
 		} else {
-			klog.V(2).Infof("Attach volume %q to instance %q failed with %v", volumeSource.DataDiskURI, nodeName, err)
+			klog.V(2).InfoS("Attach volume to node failed", "volumeDiskURI", volumeSource.DataDiskURI, "nodeName", nodeName, "err", err)
```
Comment (serathius) [inline pkg/volume/azuredd/attacher.go:91]:
```suggestion
			klog.V(2).InfoS("Attach volume to node failed", "URI", volumeSource.DataDiskURI, "node", klog.KRef("", nodeName), "err", err)
```

File: pkg/volume/azuredd/attacher.go [hunk 20]
```diff
@@ -72,23 +72,23 @@ func (a *azureDiskAttacher) Attach(spec *volume.Spec, nodeName types.NodeName) (
...
+		klog.V(2).InfoS("Attach operation is successful. Volume is already attached to node at lun.", "volumeDiskName", volumeSource.DiskName, "nodeName", nodeName, "lunNumber", lun)
 	} else {
-		klog.V(2).Infof("GetDiskLun returned: %v. Initiating attaching volume %q to node %q.", err, volumeSource.DataDiskURI, nodeName)
+		klog.V(2).InfoS("Initiating attaching volume to node. GetDiskLun returned error.", "volumeDiskURI", volumeSource.DataDiskURI, "nodeName", nodeName, "err", err)
 
 		isManagedDisk := (*volumeSource.Kind == v1.AzureManagedDisk)
 		lun, err = diskController.AttachDisk(isManagedDisk, volumeSource.DiskName, volumeSource.DataDiskURI, nodeName, compute.CachingTypes(*volumeSource.CachingMode))
 		if err == nil {
-			klog.V(2).Infof("Attach operation successful: volume %q attached to node %q.", volumeSource.DataDiskURI, nodeName)
+			klog.V(2).InfoS("Attach operation successful: volume attached to node.", "volumeDiskURI", volumeSource.DataDiskURI, "nodeName", nodeName)
```
Comment (serathius) [inline pkg/volume/azuredd/attacher.go:89]:
```suggestion
			klog.V(2).InfoS("Attach operation successful: volume attached to node.", "URI", volumeSource.DataDiskURI, "node", klog.KRef("", nodeName))
```

File: pkg/volume/azuredd/attacher.go [hunk 21]
```diff
@@ -72,23 +72,23 @@ func (a *azureDiskAttacher) Attach(spec *volume.Spec, nodeName types.NodeName) (
...
+			"nodeName", nodeName, "err", err)
 	}
 
 	if err == nil {
 		// Volume is already attached to node.
-		klog.V(2).Infof("Attach operation is successful. volume %q is already attached to node %q at lun %d.", volumeSource.DiskName, nodeName, lun)
+		klog.V(2).InfoS("Attach operation is successful. Volume is already attached to node at lun.", "volumeDiskName", volumeSource.DiskName, "nodeName", nodeName, "lunNumber", lun)
 	} else {
-		klog.V(2).Infof("GetDiskLun returned: %v. Initiating attaching volume %q to node %q.", err, volumeSource.DataDiskURI, nodeName)
+		klog.V(2).InfoS("Initiating attaching volume to node. GetDiskLun returned error.", "volumeDiskURI", volumeSource.DataDiskURI, "nodeName", nodeName, "err", err)
```
Comment (serathius) [inline pkg/volume/azuredd/attacher.go:84]:
```suggestion
		klog.V(2).InfoS("Initiating attaching volume to node. GetDiskLun returned error.", "URI", volumeSource.DataDiskURI, "node", klog.KRef("", nodeName), "err", err)
```

File: pkg/volume/vsphere_volume/attacher.go (hunk 4 above)
Comment (wangyx1992) [reply pkg/volume/vsphere_volume/attacher.go]: Hi, @serathius. The specific path such as `volumePath` are all replaced with common `path`.

File: pkg/volume/vsphere_volume/attacher.go (hunk 9 above)
Comment (wangyx1992) [reply pkg/volume/vsphere_volume/attacher.go]: Thanks for your suggestion, we replace `nodeName` with `klog.KRef("", string(nodeName))`, for that variable  `nodeName` belongs to `types.NodeName`.

File: pkg/volume/nfs/nfs.go (hunk 12 above)
Comment (wangyx1992) [reply pkg/volume/nfs/nfs.go]: Yeah,  `isMountPoint` is more readable for common users. This has been modified according to the suggestion.

File: pkg/volume/cinder/attacher_test.go (hunk 13 above)
Comment (wangyx1992) [reply pkg/volume/cinder/attacher_test.go]: Hi, @serathius. Looking up for variable `expected`, it is defined as `type disksAreAttachedCall struct`. In this struct, `ret` is defined as `error`. Can we modify `ret` to `err`?

File: pkg/volume/azuredd/azure_common_windows.go (hunk 14 above)
Comment (wangyx1992) [reply pkg/volume/azuredd/azure_common_windows.go:86]: Hi, @serathius. `lun` is the shortname for `Logical Unit Number` in Storage Area Network (SAN).

File: pkg/volume/azuredd/attacher.go (hunk 21 above)
Comment (wangyx1992) [reply pkg/volume/azuredd/attacher.go]: The above problems have been modified, waiting for your review with pleasure. @serathius

File: pkg/volume/azuredd/azure_common_windows.go (hunk 14 above)
Comment (serathius) [reply pkg/volume/azuredd/azure_common_windows.go:86]: Let's use full name in logs Logical Unit Number and capitalized key LUN as this is an acronym.

File: pkg/volume/azuredd/attacher.go [hunk 22]
```diff
@@ -72,23 +72,23 @@ func (a *azureDiskAttacher) Attach(spec *volume.Spec, nodeName types.NodeName) (
 	lun, err := diskController.GetDiskLun(volumeSource.DiskName, volumeSource.DataDiskURI, nodeName)
 	if err == cloudprovider.InstanceNotFound {
 		// Log error and continue with attach
-		klog.Warningf(
-			"Error checking if volume is already attached to current node (%q). Will continue and try attach anyway. err=%v",
-			nodeName, err)
+		klog.InfoS(
+			"Error checking if volume is already attached to current node. Will continue and try attach anyway",
+			"nodeName", nodeName, "err", err)
```
Comment (serathius) [inline pkg/volume/azuredd/attacher.go:77]: Please apply to other usage of nodeName key
```suggestion
			"node", klog.KRef("", nodeName), "err", err)
```
Node is an Kubernetes object and should be reported the same way as Pod and any other object, for example `klog.KRef(podNamespace, podName)`. 
On the other hand node is an non-namespaced object which means that you should normally pass an empty string as namespace `klog.KRef("", nodeName)`.

File: pkg/volume/azuredd/azure_common_windows.go (hunk 14 above)
Comment (wangyx1992) [reply pkg/volume/azuredd/azure_common_windows.go]: Hi, @serathius. We have updated `lun` in logs with `Logical Unit Number`, and replaced key `lun` with `LUN` at the same time.

File: pkg/volume/azuredd/attacher.go (hunk 22 above)
Comment (wangyx1992) [reply pkg/volume/azuredd/attacher.go]: This has been updated, waiting for your review with pleasure. @serathius

File: pkg/volume/cinder/attacher_test.go [hunk 23]
```diff
@@ -470,7 +470,7 @@ func (testcase *testcase) AttachDisk(instanceID, volumeID string) (string, error
 		return "", errors.New("unexpected AttachDisk call: wrong instanceID")
 	}
 
-	klog.V(4).Infof("AttachDisk call: %s, %s, returning %q, %v", volumeID, instanceID, expected.retDeviceName, expected.ret)
+	klog.V(4).InfoS("AttachDisk call volumeID and instanceID, returns retDeviceName and ret", "volumeID", volumeID, "instanceID", instanceID, "retDeviceName", expected.retDeviceName, "ret", expected.ret)
```
Comment (serathius) [inline pkg/volume/cinder/attacher_test.go:473]: Using klog in unittest doesn't seem to be correct. During tests logs should be captured and  so they can be written only when tests fails. I don't think this is supported with such calls to klog, that's why author used `klog.V(4)` to hide logs by default. Overall it's not correct to migrate those logs to InfoS as they should not be using klog at all.

Please change original calls from `klog.V(4).Infof` to `testcase.t.Logf`

Comment (wangyx1992) [reply pkg/volume/cinder/attacher_test.go]: Hi, @serathius. The  `klog.V(4).Infof` in file `attacher_test.go` have been replaced with `testcase.t.Logf`. Waiting for your review with pleasure.

**Pull request closed without merging**